bigchaindb/config_utils.py

```diff
@@ -57,13 +86,75 @@ def file_config(filename=None):
         filename = CONFIG_DEFAULT_PATH
 
     with open(filename) as f:
-        newconfig = json.load(f)
+        config = json.load(f)
 
-    dict_config(newconfig)
     logger.info('Configuration loaded from `{}`'.format(filename))
 
+    return config
+
+
+def env_config(config):
+    """Return a new configuration with the values found in the environment.
+
+    The function recursively iterates over the config, checking if there is
+    a matching env variable. If an env variable is found, the func updates
+    the configuration with that value.
+
+    The name of the env variable is built combining a prefix (``BIGCHAINDB``)
+    with the path to the value. If the ``config`` in input is:
+    ``{'database': {'host': 'localhost'}}``
+    this function will try to read the env variable ``BIGCHAINDB_DATABASE_HOST``.
+    """
+
+    def load_from_env(value, path):
+        var_name = CONFIG_SEP.join([CONFIG_PREFIX] + list(map(lambda s: s.upper(), path)))
+        return os.environ.get(var_name, value)
 
-def dict_config(newconfig):
+    return map_leafs(load_from_env, config)
+
+
+def update_types(config):
+    """Return a new configuration where all the values types
+    are aligned with the ones in the default configuration"""
+
```

Comment:
Does [this](https://github.com/bigchaindb/bigchaindb/pull/153/files#diff-377d993ccb19fa21dd579a661ff1c411R96) help? I added an explicit test.